content/blog/real-time-smart-city-traffic-monitoring-using-microservices-based-streaming-architecture-part-2.md

```diff
@@ -0,0 +1,331 @@
+
+The consumer reads the data from the stream. The tutorial code in Java from the documentation is perfectly adequate. Jackson is used to convert the JSON to Measurement objects.
+
+The consumer has an instance of the KieSession, and each measurement is added to the session using `kieSession.insert(fact)` and followed by a call to `kieSession.fireAllRules()`, which triggers the algorithm to check if any rules match with the new state of the session given the new data.
+
+A channel, which is just a callback, is used to allow a rule to call a function “outside” of the `KieSession`. My Prototype uses this method to log the alert. In a production system, we could easily change the code to send an email, an SMS, or take some other action.
+
+![](/img/picture11.png)
+
+**Importing the Rules into the Consumer Application**
+
+The way we get the rules into the running application is by fetching it from the Maven repository integrated into the Workbench.
+
+```python
+KieServices kieServices = KieServices.Factory.get();  
+ReleaseId releaseId = kieServices.newReleaseId( "org.mapr.demo", "smart-traffic-kjar", "1.0.0" );  
+KieContainer kContainer = kieServices.newKieContainer( releaseId );
+
+
+KieSession kieSession = kContainer.newKieSession();
+```
+
+So the question becomes, how does the call to `newReleaseId` know to fetch the artifact with our rules from the Maven repository in the Workbench?
+
+The answer is with the `~/.m2/settings.xml` file, where you add this information. We recommend to use the user mapr for everything in the sandbox, so the full path is: `/home/mapr/.m2/settings.xml`
+
+```xml
+[mapr@maprdemo .m2]$ cat settings.xml  
+<?xml version="1.0" encoding="UTF-8"?>  
+<settings>  
+   <servers>  
+       <server>  
+           <id>guvnor-m2-repo</id>  
+           <username>admin</username>  
+           <password>admin</password>  
+       </server>  
+   </servers>  
+   <profiles>  
+       <profile>  
+           <id>cep</id>  
+           <repositories>  
+               <repository>  
+                   <id>guvnor-m2-repo</id>  
+                   <name>Guvnor M2 Repo</name>  
+                   <url>**http://127.0.0.1:28080/drools-wb-webapp/maven2/**</url>  
+                   <releases>  
+                       <enabled>true</enabled>  
+            <updatePolicy>interval:1</updatePolicy>  
+                   </releases>  
+                   <snapshots>  
+                       <enabled>true</enabled>  
+            <updatePolicy>interval:1</updatePolicy>  
+                   </snapshots>  
+               </repository>  
+           </repositories>  
+       </profile>  
+   </profiles>  
+   <activeProfiles>  
+       <activeProfile>cep</activeProfile>  
+   </activeProfiles>  
+</settings>
+```
+
+The key information is in bold, which corresponds to the URL of the maven2 repository of the Drools workbench. This information can be copied and pasted from the pom.xml which can be seen by using the repository view:
```

Comment:
Add a comma after "which" in the second sentence:
![image](https://user-images.githubusercontent.com/77496261/137516907-2a4103d5-954d-40a0-9bf0-c44351c41483.png)

Reply:
done